qiita_db/test/test_study.py

```diff
@@ -875,6 +881,31 @@ def test_study_tags(self):
         self.assertEqual(study.tags, [])
         self.assertEqual(message, '')
 
+    def test_specimen_id_column_get_set(self):
+        self.assertEqual(self.study.specimen_id_column, None)
+        self.study.specimen_id_column = 'anonymized_name'
+        self.assertEqual(self.study.specimen_id_column, 'anonymized_name')
+        self.study.specimen_id_column = None
+        self.assertEqual(self.study.specimen_id_column, None)
+
+    def test_specimen_id_column_not_unique(self):
+        with self.assertRaises(qdb.exceptions.QiitaDBColumnError):
+            self.study.specimen_id_column = 'dna_extracted'
+
+    def test_specimen_id_column_doesnt_exist(self):
+        with self.assertRaises(qdb.exceptions.QiitaDBLookupError):
+            self.study.specimen_id_column = 'foo'
+
+    def test_specimen_id_column_no_sample_information(self):
+        empty = qdb.study.Study.create(
+            qdb.user.User('test@foo.bar'), "Fried duck microbiome",
+            self.info)
+        with self.assertRaises(qdb.exceptions.QiitaDBLookupError):
+            empty.specimen_id_column = 'foo'
+
+        # cleaning up the created study
+        qdb.study.Study.delete(empty._id)
```

Comment:
For this to always clean up, it may be worth wrapping the assertion in a try/finally